sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ParallelDocumentQueryExecutionContextBase.java

```diff
@@ -73,12 +74,17 @@ protected void initialize(
 
                 PartitionKeyInternal partitionKeyInternal = null;
                 if (cosmosQueryRequestOptions.getPartitionKey() != null && cosmosQueryRequestOptions.getPartitionKey() != PartitionKey.NONE) {
-                    partitionKeyInternal = BridgeInternal.getPartitionKeyInternal(cosmosQueryRequestOptions.getPartitionKey());
-                    headers.put(HttpConstants.HttpHeaders.PARTITION_KEY, partitionKeyInternal.toJson());
+                    // the next if statement is for the method `createDocumentServiceRequestWithFeedRange` below
+                    // with this added logic we don't have to remove the header (since the header never gets set) and
+                    // partitionKeyInternal gets passed as null which avoids the feedRange normalization.
+                    if (!PartitionKeyInternal.isPartialPartitionKeyQuery(collection, cosmosQueryRequestOptions.getPartitionKey())) {
```

Comment:
what is the harm to remove the partition key header from here completely? even for full partition key

Reply:
I'm just not trying to cause any regressions but I'm not sure if this would affect other things

Reply:
Make sense, i think this might be some left over logic when we started to use epk in query pipeline, it would be good to clean it up after this PR.